### DISCUSSION TITLE
Release 2.22 - Release Candidate

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#1778] [General]
# This is a release candidate

This is provided to collect issues & bugs and also to facilitate testing for translations.
There is a linked discussion, if you have any issues please let us know 😁 

Key changes:

* Fix BLE in PinecilV2
* Logo support for PinecilV2
* S60 Support (Limited OLED, see below)
* TS101 Support (Limited OLED, see below)
* Profile heating mode for MHP30
* Translation updates ❤️ 
* Remove 0.5 ohm tip offset to improve USB-PD compatibility
* Documentation updates
* Lots of tooling and code cleanups
* Warning if a tip is detected to be shorted (TS101 + PinecilV2)
* Filtering added to MHP tilt exit to make it less sensitive

## OLED limitations
These two new devices use a higher resolution display.
At the moment IronOS is just drawing to the top left corner. Better support is coming but is not yet finished.

## New Contributors
* @codingcatgirl made their first contribution in https://github.com/Ralim/IronOS/pull/1672
* @MAuCTEP made their first contribution in https://github.com/Ralim/IronOS/pull/1689
* @eyebrowdogs made their first contribution in https://github.com/Ralim/IronOS/pull/1715
* @TheScientistPT made their first contribution in https://github.com/Ralim/IronOS/pull/1716
* @dallinson made their first contribution in https://github.com/Ralim/IronOS/pull/1728
* @alextrical made their first contribution in https://github.com/Ralim/IronOS/pull/1729

**Full Changelog**: https://github.com/Ralim/IronOS/compare/v2.21...v2.22-rc

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.22-rc'>Release 2.22 - Release Candidate</a>.</em>

**Comment by whitequark** [comment]
Shouldn't you credit @VioletEternity for her reverse-engineering TS101?

**Reply by Ralim** [reply]
Ah heck yes definitely.
Apologies; the release notes are very largely auto generated + it being late. Will absolutely amend.

**Comment by genesi5** [comment]
Couple moments i've notice on my Pinecil V2 2.22rc:

- Under load, sometimes it swicthes off for a fractrure of second
- Tip is heating up much slower (24v PSU)
- On 2.21 i can hotswap tips, but when i do the same on 2.22rc - it wont' heat up even if i push `+`. Had to reboot iron. Not sure if it's how it should be done, but....

**Comment by Artem62500** [comment]
Also notes on S60:
- Better quality of temperature stabilization than on the original firmware (much less temperature overshoot)
- There was a high-frequency bad squeak of the PWM signal. It would be nice to bring the PWM signal out of the audible range.

**Comment by dtdionne** [comment]
Tried this on my TS101 a dozen times.  I use VMware Workstation which has dynamic Vusb mapping...just like hyperv, vbox, etc etc etc etc virtualizations.

Anyone have success with this on a TS101?

**Comment by 055dima** [comment]
As for Pinecil V2  firmware 2.22rc, 
I still have the same problems described above by the respected Genesi5
Namely, I repeat what has already been written above:
Under load, sometimes it switches off for a fraction of second
Tip is heating up much slower (28v PSU USB C 3.1)
On 2.21 i can hotswap tips, but when i do the same on 2.22rc - it wont' heat up even if i push +
I have the same problems.

**Comment by 055dima** [comment]
Regarding the soldering iron TS101 and firmware 2.22rc
First of all, I want to thank Dear @VioletEternity and Dear Ralim for the great work done.
The only thing that I do not understand is where the three Preset Temperature units T1 T2 and T3 disappeared, which were in the firmware from Miniware.
Is it just me having this problem? 
Is it possible to fix and return three Preset Temperature?
There is also a problem with the logo. 
The logo appears when power is applied to the TS101 for a microsecond and immediately disappears, despite the fact that in the TS101 settings I set the maximum logo display time.
